support/docker/build.sh

```diff
@@ -29,11 +29,136 @@ stderr.printf() {
 	printf "$@" >&2
 }
 
-_nix_docker() {
+
+_nix_container_manager() {
+
+	# Ask if the user wants to proceed as root
+	if (($(id -u) == 0)); 
+	then
+		stderr.printf "\nYou don't need to run this script as root.\n\nLear more on how run docker/podman in rootless mode:\n Docker: https://docs.docker.com/engine/security/rootless/\n Podman: https://github.com/containers/podman/blob/main/docs/tutorials/rootless_tutorial.md\n\n"
+		PS3="Running as root. Do you want to continue?: "
+
+		select yn in "Yes" "No"; do
+			case $yn in
+				Yes) 
+					stderr.printf "\nProceeding as root user\n"
+					break
+					;;
+				No)
+					stderr.printf "\nQuiting the build script...\n"
+					exit 0
+					;;
+			esac
+		done
+	fi
+
+	# TODO: Add selection menu of available boards located in boards dir
+
+	# Find which container manager is available
+
+	declare -a availableManagers=()
+
+	# Check if docker is available
+	if ! command -v docker &> /dev/null;
+	then
+		stderr.printf "Docker not installed\n"
+	else
+		availableManagers+=('docker')
+	fi
+
+
+	# Check if podman is available
+	if ! command -v podman &> /dev/null;
+	then
+		stderr.printf "Podman not installed\n"
+	else
+		availableManagers+=('podman')
+	fi
```

Comment:
We might want to add to `unavailableManagers` and print the list of unavailable managers when `availableManagers` is empty.

Otherwise we have a spurious non-actionable warning/logging printed.

Or just add to `availableManagers` in an `if` without else and not log it, since the error message currently lists the supported options.